File: Capítulo_5/Fila/Verificando se a Fila está vazia e o tamanho/isEmpty.js
// isEmpty(): esse método devolve true se a fila não contiver nenhum elemento, e false se o tamanho for maior que 0.

// size(): esse método devolve o número de elementos contidos na fila. É semelhante à propriedade length do array.
class Queue {
    constructor() {
        this.count = 0;
        this.lowestCount = 0;
        this.items = {};
    }

    isEmpty() {
        return this.count - this.lowestCount === 0; // Para calcular quantos elementos há na fila, basta calcular a diferença entre as chaves count e lowestCount.
    }

    // Assim, para implementar o método size, basta devolver esta diferença:
    size() {
        return this.count - this.lowestCount;
    }
    // Também podemos escrever método isEmpty assim:
    isEmpty() {
        return this.size() === 0;
    }
}

File: Capítulo_5/Fila/Verificando se a Fila está vazia e o tamanho/Código em Python/isEmpty.py
class Queue:
    def __init__(self):
        self.count = 0
        self.lowest_count = 0
        self.items = {}

    def is_empty(self):
        return self.count - self.lowest_count == 0
    
    def size(self):
        return self.count - self.lowest_count
    